core/src/main/scala/io/projectglow/sql/util/ExpectsGenotypeFields.scala

```diff
@@ -21,36 +21,58 @@ import org.apache.spark.sql.catalyst.analysis.{TypeCheckResult, UnresolvedExcept
 import org.apache.spark.sql.catalyst.expressions.{Expression, Unevaluable}
 import org.apache.spark.sql.types.{ArrayType, DataType, StructField, StructType}
 
+case class GenotypeInfo(size: Int, requiredFieldIndices: Seq[Int], optionalFieldIndices: Seq[Int])
```

Comment:
Can you add a brief comment documenting what these fields are? Esp. that if any indices are `-1`, they are considered missing.